packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapHeatmapController.m

```diff
@@ -0,0 +1,159 @@
+
+#import "GoogleMapHeatmapController.h"
+#import "JsonConversions.h"
+@import GoogleMapsUtils;
+
+@implementation FLTGoogleMapHeatmapController {
+  GMUHeatmapTileLayer *_heatmap;
+  GMSMapView *_mapView;
+}
+- (instancetype)initWithHeatmap:(GMUHeatmapTileLayer *)heatmap mapView:(GMSMapView *)mapView {
+  self = [super init];
+  if (self) {
+    _heatmap = heatmap;
+    _mapView = mapView;
+
+    // Workaround for weird heatmap rendering compared to other platforms
+    // Context:
+    // https://github.com/googlemaps/google-maps-ios-utils/issues/143#issuecomment-343850029
+    _heatmap.minimumZoomIntensity = 0;
+    _heatmap.maximumZoomIntensity = 21;
+  }
+  return self;
+}
+
+- (void)removeHeatmap {
+  _heatmap.map = nil;
+}
+
+- (void)clearTileCache {
+  [_heatmap clearTileCache];
+}
+
+#pragma mark - FLTGoogleMapHeatmapOptionsSink methods
+
+- (void)setWeightedData:(NSArray<GMUWeightedLatLng *> *)weightedData {
+  _heatmap.weightedData = weightedData;
+}
+
+- (void)setGradient:(GMUGradient *)gradient {
+  _heatmap.gradient = gradient;
+}
+
+- (void)setOpacity:(double)opacity {
+  _heatmap.opacity = opacity;
+}
+
+- (void)setRadius:(int)radius {
+  _heatmap.radius = radius;
+}
+
+- (void)setMap {
+  _heatmap.map = _mapView;
+}
+@end
+
+static int ToInt(NSNumber *data) { return [FLTGoogleMapJsonConversions toInt:data]; }
+
+static double ToDouble(NSNumber *data) { return [FLTGoogleMapJsonConversions toDouble:data]; }
+
+static NSArray<GMUWeightedLatLng *> *ToWeightedData(NSArray *data) {
+  return [FLTGoogleMapJsonConversions toWeightedData:data];
+}
+
+static GMUGradient *ToGradient(NSArray *data) {
+  return [FLTGoogleMapJsonConversions toGradient:data];
+}
+
+static void InterpretHeatmapOptions(NSDictionary *data, id<FLTGoogleMapHeatmapOptionsSink> sink,
+                                    NSObject<FlutterPluginRegistrar> *registrar) {
+  NSArray *weightedData = data[@"data"];
+  if (weightedData != nil) {
+    [sink setWeightedData:ToWeightedData(weightedData)];
+  }
+
+  NSArray *gradient = data[@"gradient"];
+  if (gradient != nil) {
+    [sink setGradient:ToGradient(gradient)];
+  }
+
+  NSNumber *opacity = data[@"opacity"];
+  if (opacity != nil) {
+    [sink setOpacity:ToDouble(opacity)];
+  }
+
+  NSNumber *radius = data[@"radius"];
+  if (radius != nil) {
+    [sink setRadius:ToInt(radius)];
+  }
+
+  // The map must be set each time for options to update
+  [sink setMap];
+}
+
+@implementation FLTHeatmapsController {
+  NSMutableDictionary *_heatmapIdToController;
+  FlutterMethodChannel *_methodChannel;
+  NSObject<FlutterPluginRegistrar> *_registrar;
+  GMSMapView *_mapView;
+}
+- (instancetype)init:(FlutterMethodChannel *)methodChannel
+             mapView:(GMSMapView *)mapView
+           registrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  self = [super init];
+  if (self) {
+    _methodChannel = methodChannel;
+    _mapView = mapView;
+    _heatmapIdToController = [NSMutableDictionary dictionaryWithCapacity:1];
```

Comment:
Why do we need to specifically set the initially capacity to 1. As we don't know how many controllers will be added to this map, I'd prefer use the common constructor: 
_heatmapIdToController = [[NSMutableDictionary alloc] init];

Reply:
Again based on the other controllers

Reply:
Done